mlir/include/mlir/ExecutionEngine/ROCm/BackendUitls.h

```diff
@@ -27,17 +27,29 @@ namespace mlir {
 class BackendUtils {
 public:
   BackendUtils();
+
+  // The Miir API use this constructor. Client needs to pass in the
+  // HIP property string of gcnArchName as is, in the format of arch:features
+  // i.e.: gfx908:sramecc+:xnack-
+  BackendUtils(const std::string &archName);
+
+  // mlir-rocm-runner use this constructor. It is set by the command
+  // line arguments
   BackendUtils(const std::string &triple, const std::string &chip,
                const std::string &feature);
+
   OwnedBlob compileISAToHsaco(const std::string isa, Location loc,
```

Comment:
Since I'm looking at this - are we deliberately taking a string by value here to signify a transfer of ownership?